server/webhook/merge_request.go

```diff
@@ -33,8 +33,15 @@ func (w *webhook) handleDMMergeRequest(event *gitlab.MergeEvent) ([]*HandleWebho
 		case actionReopen:
 			message = fmt.Sprintf("[%s](%s) reopen your merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
 		case actionUpdate:
-			// TODO not enough check (opened/update) to say assigned to you...
-			message = fmt.Sprintf("[%s](%s) assigned you to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+			if event.ObjectAttributes.OldRev != "" {
+				message = fmt.Sprintf("[%s](%s) pushed a new commit to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+			} else {
+				prevAssignees := event.Changes.Assignees.Previous
+				currAssignees := event.Changes.Assignees.Current
+				if len(prevAssignees) != len(currAssignees) || !compareAssignees(prevAssignees, currAssignees) {
+					message = fmt.Sprintf("[%s](%s) updated the assignees of merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+				}
+			}
```

Comment:
Some initial thoughts on this:

Posting that a new commit is on a PR can be a bit noisy, so we want to make sure a given subscription is expecting this kind of message to come in. I personally don't think it's relevant enough to warrant a DM notification.

Also, we would still want to support the "assigned you to merge request" message when a given user is added as an assignee. That's one of the most important kinds of DM notification imo

Reply:
Updated the code to support messages for removing/adding assignees. It should work fine. 
![image](https://github.com/mattermost/mattermost-plugin-gitlab/assets/44951270/07727822-331b-4b16-bca1-c4444cc57d7e)